@xen-orchestra/backups/_runners/VmsRemote.mjs

```diff
@@ -110,23 +110,27 @@ export const VmsRemote = class RemoteVmsBackupRunner extends Abstract {
                   taskByVmId[vmUuid] = new Task(taskStart)
                 }
                 const task = taskByVmId[vmUuid]
+                // error has to be caught in the task to prevent its failure, but handled outside the task to execute another task.run()
+                let taskError
                 return task
-                  .run(async () => {
-                    try {
-                      const result = await vmBackup.run()
-                      task.success(result)
-                      return result
-                    } catch (error) {
-                      if (isLastRun) {
-                        throw error
-                      } else {
-                        Task.warning(`Retry the VM mirror backup due to an error`, {
-                          attempt: nTriesByVmId[vmUuid],
-                          error: error.message,
-                        })
-                        queue.add(vmUuid)
-                      }
+                  .runInside(async () =>
+                    vmBackup.run().catch(error => {
+                      taskError = error
+                    })
```

Comment:
What about using `task.start` instead?
I personally find it more readable.
```js
// NOT TESTED
...
if (vmBackups.length > 0) {
  if (taskByVmId[vmUuid] === undefined) {
    taskByVmId[vmUuid] = new Task(taskStart)
    taskByVmId[vmUuid].start()
  }
  const task = taskByVmId[vmUuid]

  try {
    const result = await vmBackup.run()
    return task.success(result)
  } catch (error) {
    if(isLastRun) {
      return task.failure(error)
    }
    task.warning(`Retry the VM mirror backup due to an error`, {
      attempt: nTriesByVmId[vmUuid],
      error: taskError.message,
    })
    queue.add(vmUuid)
  }
  ```

Reply:
I don't think we can do like this, because if `vmBackup.run()` doesn't run in a task, the `Task.warning(...)` that occur during `vmBackup.run()` will be lost